Scripts/aws_stack_provision.ps1

```diff
@@ -1,103 +1,302 @@
 
 $EXISTING_INSTANCE_COUNT = (((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId).count
 
 if ($EXISTING_INSTANCE_COUNT -eq 1 ){
-  echo "VM with same lansa Version tag exist"
-  echo "Checking VM Status"
-  $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
-  $INSTANCE_STATE = (((Get-EC2InstanceStatus -IncludeAllInstance $true -InstanceId $INSTANCE_ID).InstanceState).Name).Value
-
-  if ($INSTANCE_STATE -eq "running"){
-     echo "VM is already in running state"}
-  elseif ($INSTANCE_STATE -eq "stopped"){
-     echo "VM is in stopped state"
-     echo "Starting the VM"
-     $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
-     Start-Sleep -Seconds 60
-     $NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
-     echo "Current status of VM is $NEW_STATUS"}
-  else{
-     echo "VM is neither Running nor stopped"
-     echo "VM status is $INSTANCE_STATE"}  }
+  
+	Write-Host "VM with Lansa Version tag = $lansa_version exist"
+	Write-Host "Checking VM Status"
+	$INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
+	$INSTANCE_STATE = (((Get-EC2InstanceStatus -IncludeAllInstance $true -InstanceId $INSTANCE_ID).InstanceState).Name).Value
+
+  	if ($INSTANCE_STATE -eq "running"){
+		Write-Host "VM is already in running state"}
+
+	elseif ($INSTANCE_STATE -eq "stopped"){
+     		Write-Host "VM is in stopped state"
+     		Write-Host "Starting the VM"
+     		$INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
+     		$NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
+
+     		Write-Host "Current status of VM is $NEW_STATUS"}
 
+  	else{
+     		Write-Host "VM is neither Running nor stopped"
+     		Write-Host "VM status is $INSTANCE_STATE"}  
+}
 elseif ($EXISTING_INSTANCE_COUNT -eq 0){
-  echo "No Exisiting VM found for this lansa version"
-  $NO_OF_AMIS = (Get-EC2Image -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version } | Select-Object ImageId | Measure-Object | Select-Object Count).count
-  if ($NO_OF_AMIS -eq 1){
-    echo "Found 1 AMI with this version tag"
-    $AMI_ID = (Get-EC2Image -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).ImageId
-    echo $AMI_ID
-    New-CNFStack -StackName .........................}
-  elseif ($NO_OF_AMIS -eq 0){
-    echo "Did not find any AMI with this tag"}
-  else{
-    echo "Found more than 1 AMI with this version tag"}}
+  	Write-Host "No Exisiting VM with Lansa Version tag = $lansa_version found"
+  	$NO_OF_AMIS = (Get-EC2Image -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version } | Select-Object ImageId | Measure-Object | Select-Object Count).count
```

Comment:
The images you look for are $clone_lansa_Version. ALWAYS.

Reply:
but we are cloning just because we dont have the AMI with the $clone_lansa_version and once we provision VM with the lansa_version we would just need to tag the VM to clone_lansa_version ?

Reply:
Have a look at the Upgrade text in the Wiki. Once you have provisioned, say, Stack 150060 using 150050 saved state, then the Architect creates the saved state for the stack: 
a) creates a new AMI and tags it with 150060
b) backs up all the databases which are tagged 150060.

Now with that Saved State for 150060, when the pipeline runs the stack provisioning scripts it ONLY passes $lansa_version = 150060 and all the Saved State exists in case it needs to create the resources again. They should be there (from the cloning process) and they may not be.